libsolidity/formal/SMTEncoder.cpp

```diff
@@ -1927,11 +1927,17 @@ smtutil::Expression SMTEncoder::compoundAssignment(Assignment const& _assignment
 
 	auto decl = identifierToVariable(_assignment.leftHandSide());
 
+	TypePointer commonType = Type::commonType(
+		_assignment.leftHandSide().annotation().type,
+		_assignment.rightHandSide().annotation().type
+	);
+	solAssert(commonType == _assignment.annotation().type, "");
+
 	if (compoundToBitwise.count(op))
 		return bitwiseOperation(
 			compoundToBitwise.at(op),
-			decl ? currentValue(*decl) : expr(_assignment.leftHandSide()),
-			expr(_assignment.rightHandSide()),
+			decl ? currentValue(*decl) : expr(_assignment.leftHandSide(), _assignment.annotation().type),
+			expr(_assignment.rightHandSide(), _assignment.annotation().type),
```

Comment:
@mijovic I think the two `expr` calls in `auto values = ...` should also be changed to be safe?

Reply:
True that. Will update them as well

Reply:
Fixed them